### DISCUSSION TITLE
Allow for multiples "openrouter.ai" in model selection

### DISCUSSION THREAD

**Discussion by sulth1** [logancyang/obsidian-copilot#368] [Ideas]
Hi Logan,

Again, thank you for your amazing work.

As more and more models are released, especially now with Opus, Sonnet and Haiku, it would be great to have the possibility to have multiple OpenRouter models (for instance, "openrouter.ai model 1", "openrouter.ai model 2", "openrouter.ai model 3"). 

That would allow the user to easily switch models on openrouter without having to go in the settings, paste the other model, save, reload, and so on. Some tasks can be done with cheaper models, while other need more sophisticated ones, so being able to quickly switch would be a great quality of life addition.

![image](https://github.com/logancyang/obsidian-copilot/assets/156309844/eb91f9aa-39ad-4afb-8cca-ab482cacb087)

![image](https://github.com/logancyang/obsidian-copilot/assets/156309844/a3d63bdc-ed78-425a-adb0-01e48358613f)

Thank you! Have a great day.

**Comment by logancyang** [comment]
Gotcha. If anyone thinks this is a good addition please upvote!

**Reply by coolk8** [reply]
Hi! Thank you for this awesome plugin!

I think the best way to customize this thing in the current LLM world is with a custom list of models, where you specify specs for every model. So it's model-based, not provider-based.

For example, this is how you create a config for models in the VS plugin "Continue":

```
    {
      "title": "openrouter/gpt-4o",
      "provider": "openai",
      "model": "openai/gpt-4o",
      "apiBase": "https://openrouter.ai/api/v1",
      "apiType": "openai",
      "contextLength": 32768,
      "apiKey": "",
      "completionOptions": {
        "maxTokens": 4096
      }
    },
    {
      "title": "openrouter/gpt-4-turbo",
      "provider": "openai",
      "model": "openai/gpt-4-turbo",
      "apiBase": "https://openrouter.ai/api/v1",
      "apiType": "openai",
      "contextLength": 32768,
      "apiKey": "",
      "completionOptions": {
        "maxTokens": 4096
      }
    },
```
You can also specify system prompt, temperature, etc. per model in this case.
I understand this is not so user-friendly, but maybe there is a possibility to do this for advanced users.
